R/htmlwidgets.R

```diff
@@ -228,18 +228,56 @@ toHTML <- function(x, standalone = FALSE, knitrOptions = NULL) {
 
 }
 
+widget_html_wrapper <- function(name, package, id, style, class, inline = FALSE, ...) {
 
-widget_html <- function(name, package, id, style, class, inline = FALSE, ...){
+  fn_res <- widget_html(
+    name = structure(name, class = c("character", name)),
```

Comment:
Sorry, I just noticed this little detail, give me a second to get my head around this.

Reply:
AFAIK s3 methods are deployed based on the class of the object into UseMethod. This sets that. Otherwise we need to do our own s3 deployment method.  unless there is another way to decide which s3 method to use?

Reply:
The problem with setting the classname on the name is that it's a lie; we shouldn't tag objects with class names unless they actually are objects of that class. I didn't look closely enough at this initially and just assumed an `x` object was being passed in and that would be what we use for S3 dispatch. I had forgotten that this `widget_html` can be called before any `x` object exists--namely, in `shinyWidgetOutput`, where we're just generating the markup which won't actually be populated with the widget until the server renders it.

So that being the case, I don't think it makes sense to use actual S3 dispatch, I'm fine with the pseudo-S3 approach.

Sorry for the unnecessary extra work. The first commit was pretty close, I'll cherry-pick from this PR and probably file a new one to hopefully be merged next week.

Reply:
No worries, It was a good thought exercise as I went through this. I wasn't super clear about my intention for a "pseudo" s3 method implementation, just that I thought something similar might be more "R-like" and less likely to introduce someone stumbling across the *widgetname*_html method and getting weird results. Thanks for the review Joe!